### PR TITLE
221 Autogen subdomain

### PR DESCRIPTION
**Pull request by BSick7**
This PR adds a data source and resource to the terraform provider:
- `data.ns_autogen_subdomain` - Used to read information about autogenerated subdomain in Nullstone
- `ns_autogen_subdomain_delegation` - Used to configure delegation set for autogenerated subdomain in Nullstone

### CODE REVIEW

File: internal/provider/config_utils.go [hunk 1]
```diff
@@ -19,3 +19,24 @@ func boolFromConfig(config map[string]tftypes.Value, key string) bool {
 	config[key].As(&val)
 	return val
 }
+
+func stringSliceFromConfig(config map[string]tftypes.Value, key string) ([]string, error) {
```
Comment (ssickles) [inline internal/provider/config_utils.go:23]: Could there be a better name for this? Slice is the operation it is doing internally but what is it actually returning? It's hard to know what this does without grokking all the code.

File: website/docs/d/autogen_subdomain.markdown [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+```
+
+## Argument Reference
+
+- `name` - (Required) Name of auto-generated subdomain that already exists in Nullstone system. This should not include `nullstone.app`.
+
+## Attributes Reference
+
+* `domain_name` - The name of the domain that Nullstone administers for this auto-generated subdomain.
```
Comment (ssickles) [inline website/docs/d/autogen_subdomain.markdown:34]: I don't see this in the example above, is this still an attribute?

**Review by ssickles — COMMENTED** [review]
I made a couple comments as a part of a quick review. I'd like to spend a couple minutes and walk through this with you.

File: internal/provider/config_utils.go (hunk 1 above)
Comment (BSick7) [reply internal/provider/config_utils.go:23]: It's returning a "string slice" from the config map.

File: website/docs/d/autogen_subdomain.markdown (hunk 2 above)
Comment (BSick7) [reply website/docs/d/autogen_subdomain.markdown:34]: It is still an attribute.
I included for future proofing; somebody would have to strip fqdn in terraform to get this value which is pretty ugly.

**Comment by BSick7** [comment]
Merging to try it out.
Can address comments in post review.

**Pull request merged**